Makefile

```diff
@@ -0,0 +1,23 @@
+.PHONY: lint
+lint:
+	flake8
+
+.PHONY: type-check
+type-check:
+	mypy .
+
+.PHONY: test
+test:
+	pytest .
+
+.PHONY: integration-test
+integration-test:
+ifndef ALERTDB_TEST_GCP_PROJECT
+	$(error "ALERTDB_TEST_GCP_PROJECT environment variable must be set for integration tests")
+endif
+
+.PHONY: precommit
+precommit: lint type-check test
```

Comment:
I don't know how you feel about it, but we're also really into the [pre-commit](https://pre-commit.com) tool at SQuaRE. Here's are some pre-commit hooks we run by default: https://github.com/lsst/templates/blob/master/project_templates/fastapi_safir_app/example/.pre-commit-config.yaml

We don't have mypy in pre-commit (though you certainly could). We end up running those through separate tox enviroments.

Reply:
Oh, nice. These are pretty slick! I've switched towards them, and dropped `make` entirely. Thanks!